src/compiler/checker.ts

```diff
@@ -21106,7 +21106,15 @@ namespace ts {
             }
 
             if (isPartOfTypeNode(node)) {
-                return getTypeFromTypeNode(<TypeNode>node);
+                let typeFromTypeNode = getTypeFromTypeNode(<TypeNode>node);
+
+                if (typeFromTypeNode && isExpressionWithTypeArgumentsInClassImplementsClause(node)) {

```

Comment:
Is this the right place to do the this instantiation or should I moce this logic into `getTypeFromTypeNode`?

I created a circularity (ie: stack overflow) when I moved it even further into the checker (namely by doing the instantiation at `getTypeFromTypeReference`).